eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -698,6 +716,155 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, val
+            remoteAmount = msg.fundingContribution,
+            commonInput_opt = Some(parentCommitments.commitment.commitInput),
+            fundingPubkeyScript = d.metaCommitments.main.commitInput.txOut.publicKeyScript, // same pubkey script as before
+            lockTime = init.lockTime,
+            dustLimit = d.metaCommitments.main.localParams.dustLimit.max(d.metaCommitments.main.remoteParams.dustLimit),
+            targetFeerate = init.feerate,
+            requireConfirmedInputs = RequireConfirmedInputs(forLocal = false, forRemote = false), // TODO: revisit this
+            localSpliceOut_opt = cmd.spliceOut_opt.map(s => TxOut(s.amount, s.pubKeyScript)),
+            remoteSpliceOut_opt = msg.spliceOut_opt
+          )
+          val txBuilder = context.spawnAnonymous(InteractiveTxBuilder(
+            nodeParams, fundingParams,
+            commitmentParams = d.metaCommitments.params,
+            purpose = InteractiveTxBuilder.SpliceTx(parentCommitments.common, parentCommitments.commitment),
+            localPushAmount = cmd.pushAmount, remotePushAmount = msg.pushAmount,
+            wallet
+          ))
+          txBuilder ! InteractiveTxBuilder.Start(self)
+          stay() using d.copy(spliceStatus = SpliceStatus.SpliceInProgress(txBuilder))
+        case _ =>
+          log.info("ignoring unexpected tx_ack_rbf")
+          stay() sending Warning(d.channelId, UnexpectedInteractiveTxMessage(d.channelId, msg).getMessage)
+      }
+
+    case Event(msg: InteractiveTxBuilder.Response, d: DATA_NORMAL) => msg match {
+      case InteractiveTxBuilder.SendMessage(msg) => stay() sending msg
+      case InteractiveTxBuilder.Succeeded(fundingParams, fundingTx, commitments) =>
+        log.info(s"splice completed: new funding txid=${fundingTx.txId} tx=${fundingTx.signedTx_opt.getOrElse("n/a")}")
+        // We now have more than one version of the funding tx, so we cannot use zero-conf.
+        val fundingMinDepth = Funding.minDepthDualFunding(nodeParams.channelConf, commitments.localParams.initFeatures, fundingParams).getOrElse(nodeParams.channelConf.minDepthBlocks.toLong)
+        blockchain ! WatchFundingConfirmed(self, commitments.fundingTxId, fundingMinDepth)
+        val metaCommitments1 = d.metaCommitments.copy(commitments = commitments.commitment +: d.metaCommitments.commitments)
+        val d1 = d.copy(metaCommitments = metaCommitments1, spliceStatus = SpliceStatus.NoSplice)
+        fundingTx match {
+          case fundingTx: PartiallySignedSharedTransaction => stay() using d1 storing() sending fundingTx.localSigs
+          case fundingTx: FullySignedSharedTransaction => stay() using d1 storing() sending fundingTx.localSigs calling publishFundingTx(fundingParams, fundingTx)
+        }
+      case f: InteractiveTxBuilder.Failed =>
+        log.info("splice attempt failed: {}", f.cause.getMessage)
+        stay() using d.copy(spliceStatus = SpliceStatus.NoSplice) sending TxAbort(d.channelId, f.cause.getMessage)
```

Comment:
I think we need a `SpliceStatus.Aborted` state where we wait for our peer to ack our `TxAbort`, similar to what we do for funding, and we must keep rejecting other messages until they've cleanly acked the abort, otherwise we may have a state mismatch.

Reply:
Done in 42673dff0.